.github/actions/set-status/action.yml

```diff
@@ -0,0 +1,48 @@
+name: 'Set Status'
+description: 'Set status on a specified head-sha'
+inputs:
+  context:
+    description: 'Name of the status'
+    required: true
+  description:
+    description: 'Short description of the status'
+    required: false
+  state:
+    description: >-
+      State of the status with possible values of 'error', 'failure', 'pending', 'success'
+    required: true
+  sha:
+    description: 'The commit ID to add the status to'
+    required: true
+  target-url:
+    description: >-
+      Target URL to associate with this status.
+      This URL will be linked from the GitHub UI to allow users to easily see the source of the status.'
+    required: false
+runs:
+  using: 'composite'
+  steps:
+    - name: Set a commit status
+      id: set-status
+      uses: actions/github-script@98814c53be79b1d30f795b907e553d8679345975 # v6.4.0 https://github.com/actions/github-script/commit/98814c53be79b1d30f795b907e553d8679345975
+      env:
+        status_context: ${{ inputs.context }}
+        description: ${{ inputs.description }}
+        state: ${{ inputs.state }}
+        sha: ${{ inputs.sha }}
+        target_url: ${{ inputs.target-url }}
+        owner: ${{ github.event.repository.owner.login }}
+        repo: ${{ github.event.repository.name }}
+      with:
+        result-encoding: string
+        script: |
+          const { status_context, description, state, sha, target_url, owner, repo } = process.env;
```

Comment:
Should we validate the required input values before calling `createCommitStatus`? (since this is a reusable action)

Reply:
Added validations for required inputs 
<img width="890" alt="SCR-20230320-lbu" src="https://user-images.githubusercontent.com/12038441/226478476-7045f081-5423-4797-a367-53614d5e114d.png">

Reply:
Code nit -- what do you think about just returning early inside each if statements? That'd help us write better error messages, and help avoid nested statements.

```ts
const { status_context, description, state, sha, target_url, owner, repo } = process.env;

if (state.length == 0 || !["error", "failure", "pending", "success"].includes(state)) {
  core.setFailed('"state" input cannot be empty.');
  return;
}

if (!["error", "failure", "pending", "success"].includes(state)) {
  core.setFailed('"state" must be a string input with possible value of (...).');
  return;
}

if (context.length <= 0) {
  core.setFailed('"context" input cannot be empty.');
  return;
}

if (sha.length <= 0) {
  core.setFailed('"sha" input cannot be empty.');
  return;
}

if (!input_string.match(/^[0-9a-z]+$/)) {
  core.setFailed('"sha" must be an alphanumeric string.')
}

github.rest.repos.createCommitStatus({
  owner,
  repo,
  sha,
  state,
  context: status_context,
  description: description.length > 0 ? description : undefined,
  target_url: target_url.length > 0 ? target_url : undefined,
});
```

Reply:
+1 to that and also having descriptive and helpful error messages

Reply:
Synced up offline with Will and we used custom messages and removed return on each.